### PR TITLE
modified transform elements to iterate through values with no corresponding field in the type

### PR DESCRIPTION
**Pull request by roironn**

### CODE REVIEW

File: packages/adapter-utils/src/utils.ts [hunk 1]
```diff
@@ -180,6 +180,14 @@ export const transformValues = (
       )
       return _.isEmpty(transformed) ? undefined : transformed
     }
+    if (_.isPlainObject(newVal) && !strict) {
+      const transformed = _(newVal)
+        .mapValues((val, key) => transformValue(
+          val,
+          keyPathID?.createNestedID(key),
+        )).omitBy(_.isUndefined).value()
```
Comment (ori-moisis) [inline packages/adapter-utils/src/utils.ts:188]: nit, just to be consistent with the style in the rest of the function
```suggestion
      const transformed = _.omitBy(
        _.mapValues(
          newVal,
          (val, key) => transformValue(val, keyPathID?.createNestedID(key),
        ),
        _.isUndefined,
      )
```

File: packages/adapter-utils/test/utils.test.ts [hunk 2]
```diff
@@ -565,10 +569,19 @@ describe('Test utils.ts', () => {
     })
 
     describe('when strict is false', () => {
+      const unmergedValues = {
```
Comment (ori-moisis) [inline packages/adapter-utils/test/utils.test.ts:572]: what does `merge` have to do with anything?

**Review by ori-moisis — APPROVED** [review]
Seems like this should work
I hope it doesn't have too big of an effect on our performance (would cause `transformElement` to iterate on things it didn't in the past)

File: packages/adapter-utils/test/utils.test.ts (hunk 2 above)
Comment (roironn) [reply packages/adapter-utils/test/utils.test.ts:572]: this happens IRL when instances are not merged so the type is a placeholder and not the real type... Couldn't think of a better name :)

I'll figure something out.

**Comment by roironn** [comment]
> Seems like this should work
> I hope it doesn't have too big of an effect on our performance (would cause `transformElement` to iterate on things it didn't in the past)

Only in 'unstrict' mode... so I guess we should be fine?

**Comment by ori-moisis** [comment]
> > Seems like this should work
> > I hope it doesn't have too big of an effect on our performance (would cause `transformElement` to iterate on things it didn't in the past)
> 
> Only in 'unstrict' mode... so I guess we should be fine?

most transforms run in `unstrict` mode IIRC

File: packages/adapter-utils/test/utils.test.ts (hunk 2 above)
Comment (ori-moisis) [reply packages/adapter-utils/test/utils.test.ts:572]: untyped?

**Pull request merged**